[25.0] DatasetView and Card Polish

dannon opened the pull request:
Smoothing out user interactions with the new streamlined view.

xref https://github.com/galaxyproject/galaxy/issues/20301

Adjusted header:
Nicer formatting, dropped 'info', which folks agreed was usually cluttered with unhelpful stderr.  
We keep 'blurb'.
![image](https://github.com/user-attachments/assets/46662e74-248d-43c0-beb7-21bd57d21aa2)



For the card options, eye and error are dropped and we're going with a new icon.

Here's faExpand:
![image](https://github.com/user-attachments/assets/4c970098-dfc0-4127-9694-734678b64d58)

And here's ScanEye:
![image](https://github.com/user-attachments/assets/c02c8178-e160-4d0e-8851-cd227b517bb8)


I'm also going to mock up a quick composite of faEye and faExpand  (which will be a lot like ScanEye, but with fa styles) in a sec.


## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

dannon committed:
Add icons to tabs in DatasetView

dannon committed:
drop library, use icons directly

dannon committed:
Use faExpand for 'view' of dataset

dannon committed:
Drop pencil edit from dataset card (again -- we'll have to fix some tests here)

dannon committed:
Drop error from dataset card

dannon committed:
Refine DatasetView header to show essential fields only

- Remove misc_info field that is overly often filled with garbage
- Add file size display with human-readable formatting
- Improve styling with proper spacing and muted labels

dannon committed:
Tooltips for the DatasetView tabs

dannon committed:
Try out scan-eye for dataset view icon

dannon committed:
Add back dataset card icons -- remove in dev.

dannon committed:
Add Raw tab for datasets with automatic visualization

When a dataset has a preferred visualization, the Preview tab shows the
visualization. This adds a Raw tab to allow users to view the raw data
display alongside the visualization.

dannon committed:
Prevent auto-download file types from triggering downloads

Certain file types (TIFF, PDF, office documents, archives) trigger automatic
downloads when loaded in an iframe. This change detects these file types and
shows a download prompt instead of attempting to display them inline.

dannon committed:
Enhance API response for datatypes to include extra display information so the client can account for direct download vs preview, etc.

dannon committed:
Use store, cache datattypeDetails

dannon committed:
Add Raw tab for visualized datasets and prevent auto-downloads

- Add Raw tab to DatasetView when datasets have automatic visualizations,
allowing users to view raw data alongside visualizations
- Prevent file types that trigger automatic downloads (TIFF, PDF, etc.)
from loading in iframes, showing download prompt instead

dannon committed:
Rebuild api schema

dannon committed:
Update selectors for Edit, Details, and Visualize tabs to use more descriptive titles

dannon committed:
Update tab titles for navigation.yml

dannon committed:
Fix prop passing to centerpane

mvdbeek commented on client/src/api/datatypes.ts:
```diff
@@ -1,3 +1,20 @@
+ * Get details about a specific datatype
+ */
+export async function fetchDatatypeDetails(extension: string) {
+    try {
```
This is a fastapi route, any reason not to use the typed fetcher ?

mvdbeek reviewed (approved):
Nice improvements, thanks so much @dannon!

dannon committed:
Route directly to visualize tab for restored icon in datasetcard

dannon committed:
Swap datatypes info request to fetch client

dannon committed:
Wait for datatype response before setting up tabs -- we need this to know if we are preview/viz/raw/etc.

dannon committed:
Consolidate loading states, fix API imports.

dannon committed:
Fix dataset view heading wrapping and consolidate loading states

Prevent line breaks in hid and dataset state while allowing name to
wrap. Consolidate datatype and dataset loading states into single
computed property.
